### PR TITLE
Add missing Krav Maga Weapons: Take 2

### PR DESCRIPTION
**Pull request by JohnWildkins**
#### Summary
```SUMMARY: Balance "All semiauto rifles, full size semiauto pistols, and knives can be used with Krav Maga."```

#### Purpose of change
As noted in #33867, Krav Maga is currently incompatible with several weapons that are either functionally identical to ones that can be used, or similar enough that they should be usable without issue. This PR adds several (and removes one, pending discussion) weapons that meet the initial requirements or are sensible additions given the history of Krav Maga, and where and how it is employed. This change is primarily meant to reduce user confusion, where a M1911 or Glock 17 is an acceptable Krav Maga handgun, but the Beretta M9A1 is not, for example.

#### Describe the solution
Several new weapons were made valid, in a few different addition categories. From least to most controversial (already valid weapons in bold):

> **combat knife**: modified combat knife
> **switchblade**: folding knife, swiss army knife

Three knives that weren't covered by the original PR.

> **M1911**: AF2011A1, M45A1
> **USP series**: MK 23
> **L39**: L39-460 
> **L39B**: L39B-45
> **AR-15**: M4A1, M16A4
> **M14 EBR**: Ruger Mini-14, M1A
> **SCAR-L**: SCAR-H.

Pistols and rifles that are functionally identical to those already usable, typically retoolings or military variants.

> Browning Hi-Power series, AK series, IWI Tavor X95

Weapons systems that are/were in use by the IDF in some form. 

> **AR-15 style platforms**: HK416/417, Remington ACR, M110A1
> **Full-size handguns**: Walther handguns, Desert Eagle .44, CZ-75, Beretta M9A1
> _AK-style platforms_: AN-94

Weapons systems that are similar enough to accepted weapons to be reasonable additions. (The AN-94 would only be reasonable as an off-shoot of the AK, hence the special note.)

> _Removed_: M24

Under the scope provided by the initial Krav Maga rework (#33602), only (semi) automatic rifles and pistols (excl. revolvers and subcompacts) were applicable. While I don't personally agree that it should be that limited, as it stands the M24 is a black sheep as it's the only bolt-action long rifle permitted.

#### Describe alternatives you've considered
The main problem arising from this is that Krav Maga, which already had the longest acceptable weapons list in the game, now has one that is rather excessive. Keeping this list and maintaining it (now at nearly 100 entries) is rather unfeasible, and the presentation in-game is nigh unreadable. A better system might be allowing entire classes of weapons to be dictated as acceptable, i.e. anything that is considered a handgun or rifle, and then blacklisting certain weapons instead. That's beyond my expertise, however.

In addition, there's the question of which weapon classes are acceptable. There isn't much information out there beyond yes, indeed, they are used offensively and that includes both handguns and service rifles; but there are arguments to be made and edge cases as well. Many rifles (including the M110A1 included in this PR) are designated marksman rifles, not service rifles, however they're very similar in form factor to a standard rifle. The same goes for shotguns like the Saiga-410, which is almost literally an AK-pattern shotgun. Surely if an AK-47 is acceptable, than an AK-sized and AK-shaped shotgun is too? Carbines and full-sized SMGs also blur the line here. Pipe rifles and other homemade weapons are also still not included, purely because I wasn't sure where to draw the line on some. Then there's the question of things like revolvers, of which one is currently included by the base PR (the L2032), lever-action rifles, bolt-action rifles, et cetera et cetera. 

#### Testing
Tested by checking the acceptable weapons list in-game, as well as wielding a few examples and ensuring the combat style worked.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
What does the action of a weapon have to do with its applicability as a melee weapon?

**Comment by JohnWildkins** [comment]
> 
> 
> What does the action of a weapon have to do with its applicability as a melee weapon? If it has a suitable stock, it should be good to go.
> 
> If the floodgates must be opened for Krav Maga to use firearms in melee, everything that has a sufficient buttstock should be valid. Anything long enough, with a buttstock, and not exceeding... Idk, 10 pounds? Any pistol that is large enough should also be valid.

I agree wholeheartedly, frankly. I just don't think it's feasible to maintain a static string list of every gun in the game, save like three. This list was compiled based on the ground rules laid by the first two PRs, since I was frankly just getting annoyed that commonplace guns like the M9A1 weren't valid.

I might try and change the logic of martial arts to allow for weapon categories to be selected instead, but I believe the only categorization in game is done by what skill they level, which might be an equally clumsy solution.

> The Swiss army knife is TINY, I do not think this is a valid melee weapon.

It isn't _that_ small, and it already has a nasty to-hit malus and pretty poor damage. I don't think allowing it is bending feasibility, but it's not a hill I need to die on, either.

**Comment by ifreund** [comment]
Swiss army knives don't lock open so you run a risk of the blade closing on your fingers if force is applied to the back of the blade.  They are also tiny and flimsy as already stated. I don't think the swiss army knife is in any way equivalent to a switchblade or folding knife, both of which have locking blades and are more robust.

File: data/json/martialarts.json [hunk 1]
```diff
@@ -590,9 +590,12 @@
       "diveknife",
       "honey_scraper",
       "knife_combat",
+      "knife_combat_mod",
+      "knife_folding",
       "knife_hunting",
       "knife_rambo",
       "knife_rm42",
+      "knife_swissarmy",
```
Comment (kevingranade) [inline data/json/martialarts.json:598]:
```suggestion
```

**Pull request merged**